packages/core/src/shader/Shader.ts

```diff
@@ -70,8 +75,16 @@ export class Shader
         return false;
     }
 
+    /** Disposes WebGL resources that are connected to this shader. */
+    dispose(): void
+    {
+        this.disposeRunner.emit(this, false);
+    }
+
     destroy(): void
     {
+        this.dispose();
```

Comment:
You probably don't need the dispose function. And it's kind of confusing to have destroy and dispose anyways. Maybe just in-line the runner emit here?